pandas/_libs/tslibs/offsets.pyx

```diff
@@ -4635,6 +4656,8 @@ def _get_offset(name: str) -> BaseOffset:
     --------
     _get_offset('EOM') --> BMonthEnd(1)
     """
+    if name.upper() != name:
+        name = name.lower()
```

Comment:
it's necessary in order to catch frequencies like `'qe-MAr'` with mix of lower/uppercase letters. Otherwise we should add all possible combinations to `_dont_uppercase` list.

Reply:
ah I see, thanks!

Before this change:
```python
In [4]: _get_offset('Qe-mAR')
Out[4]: <QuarterEnd: startingMonth=3>
```
After this change:
```python
In [2]: _get_offset('Qe-mAR')
---------------------------------------------------------------------------
ValueError: Invalid frequency: qe-mar
```

Would it work to simplify and just do
```diff
-    if name.upper() != name:
-        name = name.lower()
-    if name not in _dont_uppercase:
+    if name.lower() not in _dont_uppercase:
```
?

Reply:
Thank you, I simplified the condition. I just wasn't sure that the frequency like `‘QE-MAR’` should also pass this check.

It looks confusing: we take an uppercase string, then lowercase it and check if the string is in the `_dont_uppercase` list.

Reply:
agree, but I think we want to get rid of `_dont_uppercase` completely anyway